devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,199 @@
+                attributeState "both", label:'ring!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#0e7507"
+            }
+        }
+        standardTile("off", "device.alarm", inactiveLabel: false, decoration: "flat") {
+            state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
+        }
+        valueTile("tamper", "device.tamper", height: 2, width: 2, decoration: "flat") {
+            state "clear", label: 'tamper clear', backgroundColor: "#ffffff"
+            state "detected", label: 'tampered', backgroundColor: "#ffffff"
+        }
+
+        main "alarm"
+        details(["alarm", "off",  "tamper"])
+    }
+}
+
+def installed() {
+    initialize()
+    sendEvent(name: "alarm", value: "off", isStateChange: true)
+}
+
+def updated() {
+    initialize()
+}
+
+def initialize() {
+    if (!childDevices) {
+        addChildren()
+    }
+    sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+}
+
+def parse(String description) {
+    def result = null
+    def cmd = zwave.parse(description)
+    if (cmd) {
+        result = zwaveEvent(cmd)
+    }
+    log.debug "Parse returned: ${result.inspect()}"
+    return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+    createEvents(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+    createEvents(cmd.value)
+}
+
+def createEvents(value) {
+    sendEvent(name: "alarm", value: value ? "both" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+    def encapsulatedCommand = cmd.encapsulatedCommand()
+    def endpoint = cmd.sourceEndPoint
+    if(endpoint == 1) {
+        zwaveEvent(encapsulatedCommand)
+    } else {
+        String childDni = "${device.deviceNetworkId}:$endpoint"
+        def child = childDevices.find { it.deviceNetworkId == childDni }
+
+        child?.handleZWave(encapsulatedCommand)
+    }
+}
+
+def on() {
+    def cmds = []
+    cmds << basicSetCmd(0xFF, 1)
+    delayBetween(cmds, 100)
+}
+
+def off() {
+    def cmds = []
+    cmds << basicSetCmd(0x00, 1)
+    delayBetween(cmds, 100)
+}
+
+def strobe() {
+    on()
+}
+
+def siren() {
+    on()
+}
+
+def both() {
+    on()
+}
+
+def ping() {
+    encap(zwave.basicV1.basicGet())
+}
+
+private addChildren() {
+    def numberOfSounds = 6
+    for(def endpoint : 2.. numberOfSounds) {
+        try {
+            String childDni = "${device.deviceNetworkId}:$endpoint"
+            def componentLabel = device.displayName[0..-2] + "$endpoint"
+
+            addChildDevice("Child Stateless Siren", childDni, device.getHub().getId(), [
+                    completedSetup: true,
+                    label: componentLabel,
+                    isComponent: false,
```

Comment:
I think this device should at least be MCD but I'm not sure that's the best way to model it in SmartThings since I don't think it will be clear to the user what the different components do. I don't recall if there is a appropriate capability for defining a supported set of notifications and letting the user select one when setting it up in an automation, but that seems like the better user experience to me.

Reply:
@tpmanley I don't think there's anything like that right now. Konrad is basically redoing the "ecolink wireless siren" DTH here, but I agree that we can do better.

For right now I think the best we can do is use to make this a multi-component device and to use the regular Alarm capability instead of the stateless siren one, since the device can be configured to update us when tones are done playing:

https://products.z-wavealliance.org/products/3218

Reply:
@greens thing is, even though we get notification that tune has ended, we cannot determine on which endpoint it was playing, we just receive NotificationReport on start & end, but there's no way to identify exact sound depending on that report. That's why I've suggested Michał to use stateless capability, cause while it's doable with standard Alam cap, it'd include some scheduling tricks and put unnecessary stress on cloud and imo it's not worth it, even so it probably won't be widely used device.

Reply:
@PKacprowiczS I wouldn't necessarily make assumptions about if the device is widely used (you may be surprised).

I assume that only one sound can play at a time? If so, just keep track of the last sound playing. In fact, the children should be keeping track of that for you -- just check for the child device that has device.currentValue("alarm") != "off".

An alternate method to do this is like what I did in the Philio Multi-Sound Siren (see https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/philio-multiple-sound-siren.src/philio-multiple-sound-siren.groovy). Have a device preference for the sound that you want, and use Alarm and Chime.

A few other suggestions:
If we go the route of MCD, I would use translatable component labels instead of ones that have the component number. I can send a link to some code that is in the private repo where I do this.

For the sounds that are doorbell noises, and in the parent, make use of the Chime capability, as well.